### PR TITLE
Nomicon: Vec ZST a few code fixes

### PR DESCRIPTION
**Pull request by jtepe**
- Moved semicolon to the right place in the `let` statement in the ZST section.
- Fixed the missing ZST additions for `RawValIter<T>` from this section in the final code section.

### CODE REVIEW

**Comment by Gankro** [comment]
The indentation of those `if`s seems unidiomatic. I do:

```rust
let x = if foo() {
    a
} else {
    b
}
```

But maybe the community has drifted from that?

**Comment by Gankro** [comment]
r=me on the content of the patch, though

File: src/doc/nomicon/vec-final.md [hunk 1]
```diff
@@ -226,7 +226,11 @@ impl<T> Iterator for RawValIter<T> {
         } else {
             unsafe {
                 let result = ptr::read(self.start);
-                self.start = self.start.offset(1);
+                self.start = if mem::size_of::<T>() == 0 {
+					(self.start as usize + 1) as *const _
```
Comment (bluss) [inline src/doc/nomicon/vec-final.md]: It looks like you have literal tabs in the file, you should indent with only spaces.

**Comment by jonastepe** [comment]
Damn Emacs config ...

**Pull request merged**